File: 백준/Silver/2559. 수열/수열.swift
import Foundation

let NK = readLine()!.split(separator:" ").compactMap { Int($0) },
    n = NK[0],
    k = NK[1],
    temperatures = readLine()!
    .split(separator: " ")
    .compactMap { Int($0) }
    
var sumTemperature = temperatures[0..<k].reduce(0, +),
    maxTemperature = sumTemperature

for i in (k..<n) {
    sumTemperature += temperatures[i] - temperatures[i-k]
    maxTemperature = max(maxTemperature, sumTemperature)
}

print(maxTemperature)

File: 백준/Silver/2559. 수열/수열.py
N, K = [int(x) for x in input().split()]  #

temperatures = [int(x) for x in input().split()]

sum_temperature = sum(temperatures[:K])
max_temperature = sum_temperature

for i in range(K,N):
    sum_temperature += temperatures[i]
    sum_temperature -= temperatures[i-K]
    if sum_temperature > max_temperature:
        max_temperature = sum_temperature

print(max_temperature)